contracts/FlowIDTableStaking.cdc

```diff
@@ -833,41 +878,55 @@ pub contract FlowIDTableStaking {
 
                 // Remove nodes if they do not meet the minimum staking requirements
                 if !FlowIDTableStaking.isGreaterThanMinimumForRole(numTokens: totalTokensCommitted, role: nodeRecord.role) ||
-                   (approvedNodeIDs[nodeID] == nil) {
-
-                    emit NodeRemovedAndRefunded(nodeID: nodeRecord.id, amount: nodeRecord.tokensCommitted.balance + nodeRecord.tokensStaked.balance)
-
-                    // move their committed tokens back to their unstaked tokens
-                    nodeRecord.tokensUnstaked.deposit(from: <-nodeRecord.tokensCommitted.withdraw(amount: nodeRecord.tokensCommitted.balance))
-
-                    // Set their request to unstake equal to all their staked tokens
-                    // since they are forced to unstake
-                    nodeRecord.tokensRequestedToUnstake = nodeRecord.tokensStaked.balance
-
-                    // Iterate through all delegators and unstake their tokens
-                    // since their node has unstaked
-                    for delegator in nodeRecord.delegators.keys {
-                        let delRecord = nodeRecord.borrowDelegatorRecord(delegator)
-
-                        if delRecord.tokensCommitted.balance > 0.0 {
-                            emit DelegatorTokensUnstaked(nodeID: nodeRecord.id, delegatorID: delegator, amount: delRecord.tokensCommitted.balance)
-
-                            // move their committed tokens back to their unstaked tokens
-                            delRecord.tokensUnstaked.deposit(from: <-delRecord.tokensCommitted.withdraw(amount: delRecord.tokensCommitted.balance))
-                        }
-
-                        // Request to unstake all tokens
-                        delRecord.tokensRequestedToUnstake = delRecord.tokensStaked.balance
+                   (nodeRecord.role != UInt8(5) && approvedNodeIDs[nodeID] == nil) {
+                    self.removeAndRefundNodeRecord(nodeRecord: nodeRecord)
+                } else {
+                    //capture whether this is a current node or a pending node
+                    if nodeRecord.initialWeight > 0 {
+                        currentNodeCount[nodeRecord.role] = currentNodeCount[nodeRecord.role]! + 1
+                    } else if candidateNodes!.contains(nodeID) {
+                        pendingNodes[nodeRecord.role]!.append(nodeRecord)
```

Comment:
@joshuahannan Just thinking through this and currently there is an attack surface here because people could add lots of nodes.

What if we have a hard-limit of the number of new pending nodes per epoch right now in order to limit this?